### ISSUE TITLE
ProcessStartInfo doesn't pass empty string to the target process

### ISSUE THREAD

**Issue by EmmaZhu** [dotnet/corefx#25456]
Following code is to pass "test1" and an empty string as arguments to netcoretest application. It works well on Windows. On Ubuntu (16.10 and 17.04), it just passes "test1" to netcoretest, and ignores the empty string. Could anyone help to take a look?

```
            ProcessStartInfo psi = new ProcessStartInfo("/usr/bin/dotnet");
            psi.Arguments = "netcoretest.dll test1 \"\"";

            Process p1 = Process.Start(psi);
            p1.OutputDataReceived += (s, eventArg) =>
            {
                Console.WriteLine("P1 {0}", eventArg.Data.ToString());
            };
```

**Comment by danmosemsft** [comment]
@EmmaZhu we might not be able to look for a bit. If this is impacting you, perhaps you could debug some?

**Comment by stephentoub** [comment]
I believe this was already fixed for 2.1.
https://github.com/dotnet/corefx/pull/21339

I compiled this with `gcc -o test test.c`:
```C
#include <stdio.h>

int main(int argc, char** argv)
{
    printf("argc: %d\n", argc);
    for (int i = 0; i < argc; i++) printf("argv[%d]: %s\n", i, argv[i]);
    return 0;
}
```

Then I ran this app using 2.0:
```
using System;
using System.Diagnostics;

class Program
{
    static void Main() => Process.Start("/home/stoub/tmpapp/test", "test1 \"\"");
}
```
and got:
```
stoub@SToubUbuntu17102:~/tmpapp$ dotnet run
argc: 2
argv[0]: /home/stoub/tmpapp/test
argv[1]: test1
```
but with bits from master, got:
```
stoub@SToubUbuntu17102:~/CoreClrTest$ ./corerun /home/stoub/tmpapp/bin/Debug/netcoreapp2.0/tmpapp.dll
argc: 3
argv[0]: /home/stoub/tmpapp/test
argv[1]: test1
argv[2]:
```
